lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -431,15 +432,34 @@ def current(self) -> Node:
                 temperature=current_node.get("attributes", {}).get("temp"),
             )
         except OpenSearchHttpError:
-            # we try to get the most accurate description of the node
-            conf_on_disk = self.config.load("opensearch.yml")
+            # we try to get the most accurate description of the node from the static config
+            conf = self.config.load("opensearch.yml")
+            try:
+                roles = conf["node.roles"]
+            except KeyError:
+                roles = self._charm.opensearch_peer_cm.deployment_desc().config.roles
+
+            try:
+                app = App(id=conf["node.attr.app_id"])
+            except KeyError:
+                app = App(id=self._charm.opensearch_peer_cm.deployment_desc().app.id)
+
+            try:
+                temperature = conf["node.attr.temp"]
+            except KeyError:
+                temperature = (
+                    self._charm.opensearch_peer_cm.deployment_desc().config.data_temperature
+                )
+            if not (app and roles):
```

Comment:
The app should never be None
```suggestion
            if not roles:
```